src/withApolloClient.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * @author sarahr
+ *
+ * This is intended to wrap the root of our app with the Apollo client so all our components
+ * can run graphQL queries.
+ *
+ * There are 2 ways to do this - with apollo-boost and apollo-client.
+ * apollo-boost supposedly "configures your client with recommended settings... like our
+ * in memory cache, local state management, and error handling."
+ *
+ * apollo-client however allows you to customize links, and with our implementation
+ * we need a custom link with `credentials: 'same-origin'` set.
+ */
+import React, { Component } from "react";
+import ApolloClient from "apollo-client";
+import fetch from "node-fetch";
+import { ApolloProvider } from "react-apollo";
+import { createHttpLink } from "apollo-link-http";
+import { InMemoryCache } from "apollo-cache-inmemory";
+
+const withApolloClient = (App) => {
+    return class extends Component {
+        render() {
+            const link = createHttpLink({
+                uri: "/graphql",
+                credentials: "same-origin",
+                fetch: fetch,
+            });
+            const apolloClient = new ApolloClient({
```

Comment:
It seems that this will create a new `ApolloClient` every time this `withApolloClient` component re-renders.
I'm not sure if this is required (maybe it will only render once per request or per user session?), or if we want to define 1 static `apolloClient` and `link` outside this wrapper component?

Reply:
I admit, this was my question too. The [apollo tutorial](https://www.apollographql.com/docs/react/get-started/#create-a-client) seems to create a `const` client directly in the namespace, then [passes that in](https://www.apollographql.com/docs/react/get-started/#connect-your-client-to-react) to the `ApolloProvider`. We do need to handle auth, of course. I notice that there's a function option you can configure the ApolloClient with, called [`request`](https://www.apollographql.com/docs/react/get-started/#configuration-options) that it mentions is useful for authentication purposes, that's called on every request and can be used to set `fetchOptions`. I didn't get much of a chance to dig into it beyond looking at docs, admittedly, so there may be a reason, but I admit that the approach with one client rather than a new one each render seems a lot cleaner to me.

Reply:
That's a good point.  I could move the `link` and `apolloClient` creation to outside of the wrapper.  As for auth, I think we have that settled now by using `credentials: 'same-origin'`, but that was Sarah's finding so she probably knows more about it than I do.